Document Android emulator usage

dbnicholson opened the pull request:
I've found that developing against the Android emulator results in a much more streamlined development experience. This documents how to use the emulator without Android Studio.

https://phabricator.endlessm.com/T33535

dbnicholson committed:
Add basic command line Android Emulator usage documentation

This provides a much faster development experience than installing and
running on a real Android device.

https://phabricator.endlessm.com/T33535

dbnicholson committed:
Document using an SD card image with Android Emulator

This is helpful when trying to simulate the experience with external
storage. Although the SD card isn't true removable storage like a USB
drive, it will still require access using the same Android APIs.

https://phabricator.endlessm.com/T33535

manuq commented on README.md:
```diff
@@ -114,3 +114,142 @@ You could also do so using [Weinre](https://people.apache.org/~pmuellr/weinre/do
+
+First, ensure the emulator is installed in the Android SDK. The `make setup`
+target will install the necessary pieces. Assuming the SDK is installed in
+`~/.android/sdk`:
```
From the toolbox created with `make build_toolbox` this path is `/opt/android/sdk/`.

dbnicholson replied:
Hmm, that might be better. You do need to be root to install SDK components then, but it would be more consistent.

manuq commented on README.md:
````diff
@@ -114,3 +114,142 @@ You could also do so using [Weinre](https://people.apache.org/~pmuellr/weinre/do
+the system image provided above and a Pixel 5 device:
+
+```
+~/.android/sdk/cmdline-tools/latest/bin/avdmanager create device --name test \
````
In my case from toolbox this was `create avd` not `create device`.
```suggestion
~/.android/sdk/cmdline-tools/latest/bin/avdmanager create avd --name test \
```

dbnicholson replied:
Oops, that is correct.

manuq reviewed (approved):
Looks great. This doesn't work from toolbox:

```
$ /opt/android/sdk/emulator/emulator -avd test
INFO    | Android emulator version 31.2.10.0 (build_id 8420304) (CL:N/A)
/opt/android/sdk/emulator/qemu/linux-x86_64/qemu-system-x86_64: error while loading shared libraries: libpulse.so.0: cannot open shared object file: No such file or directory
```

But I guess that's expected. Please check my suggestion, Either it's a typo or my SDK version in toolbox is not the same as the one installed by `make setup`.

dbnicholson commented:
> /opt/android/sdk/emulator/qemu/linux-x86_64/qemu-system-x86_64

That's interesting. I would have figured they'd bundle all the dependencies, but I guess not. You could try installing `qemu-system-x86` in the container. That would hopefully get you all the same package dependencies.

dbnicholson committed:
Add QEMU packages to toolbox image

The Android emulator in the SDK does not contain all the host
dependencies as these are expected to ship in Android Studio (I
believe). Install the QEMU packages from Endless as a means to get the
them. Also include qemu-utils for `qemu-img` and `qemu-nbd` even though
`qemu-nbd` won't work unless you're real root in the container.

https://phabricator.endlessm.com/T33535

dbnicholson commented:
I was able to run `emulator` in toolbox once I installed the QEMU packages, so I added those to the toolbox image.

manuq commented:
retest this please
